pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -159,435 +49,423 @@ func TestApplyRemoteConfig(t *testing.T) {
-}
-
-func TestApplyRemoteWindowsEventConfig(t *testing.T) {
-	d, err := os.MkdirTemp("", "opamp.d")
-	assert.NoError(t, err)
-	defer os.RemoveAll(d)
-
-	cfg := createDefaultConfig().(*Config)
-	cfg.RemoteConfigurationDirectory = d
-	set := extensiontest.NewNopSettings()
-	o, err := newOpampAgent(cfg, set.Logger, set.BuildInfo, set.Resource)
-	assert.NoError(t, err)
-
-	assert.Equal(t, len(o.effectiveConfig), 0)
-
-	path := filepath.Join("testdata", "opamp.d", "opamp-windows-event-config.yaml")
-	rb, err := os.ReadFile(path)
-	assert.NoError(t, err)
-
-	rc := &protobufs.AgentRemoteConfig{
-		Config: &protobufs.AgentConfigMap{
-			ConfigMap: map[string]*protobufs.AgentConfigFile{
-				"default": {
-					Body: rb,
-				},
+		{
+			name: "ApplyRemoteApacheConfig",
+			validate: func(o *opampAgent, t *testing.T) {
+				path := filepath.Join("testdata", "opamp.d", "opamp-apache-config.yaml")
+				rc := remoteConfig(t, path)
+
+				changed, err := o.applyRemoteConfig(rc)
+				assert.NoError(t, err)
+				assert.True(t, changed)
+				assert.NotEqual(t, len(o.effectiveConfig), 0)
+
+				o.cfg.AcceptsRemoteConfiguration = false
+				changed, err = o.applyRemoteConfig(rc)
+				assert.False(t, changed)
+				assert.Error(t, err)
+				assert.Equal(t, "OpAMP agent does not accept remote configuration", err.Error())
 			},
 		},
-		ConfigHash: []byte("b2b1e3e7f45d564db1c0b621bbf67008"),
-	}
-
-	changed, err := o.applyRemoteConfig(rc)
-	assert.NoError(t, err)
-	assert.True(t, changed)
-	assert.NotEqual(t, len(o.effectiveConfig), 0)
-
-	cfg.AcceptsRemoteConfiguration = false
-	changed, err = o.applyRemoteConfig(rc)
-	assert.False(t, changed)
-	assert.Error(t, err)
-	assert.Equal(t, "OpAMP agent does not accept remote configuration", err.Error())
-}
-
-func TestApplyRemoteExtensionsConfig(t *testing.T) {
-	d, err := os.MkdirTemp("", "opamp.d")
-	assert.NoError(t, err)
-	defer os.RemoveAll(d)
-
-	cfg := createDefaultConfig().(*Config)
-	cfg.RemoteConfigurationDirectory = d
-	set := extensiontest.NewNopSettings()
-	o, err := newOpampAgent(cfg, set.Logger, set.BuildInfo, set.Resource)
-	assert.NoError(t, err)
-
-	assert.Equal(t, len(o.effectiveConfig), 0)
-
-	path := filepath.Join("testdata", "opamp.d", "opamp-extensions-config.yaml")
-	rb, err := os.ReadFile(path)
-	assert.NoError(t, err)
-
-	rc := &protobufs.AgentRemoteConfig{
-		Config: &protobufs.AgentConfigMap{
-			ConfigMap: map[string]*protobufs.AgentConfigFile{
-				"default": {
-					Body: rb,
-				},
+		{
+			name: "ApplyRemoteHostConfig",
+			validate: func(o *opampAgent, t *testing.T) {
+				path := filepath.Join("testdata", "opamp.d", "opamp-host-config.yaml")
+				rc := remoteConfig(t, path)
+
+				changed, err := o.applyRemoteConfig(rc)
+				assert.NoError(t, err)
+				assert.True(t, changed)
+				assert.NotEqual(t, len(o.effectiveConfig), 0)
+
+				o.cfg.AcceptsRemoteConfiguration = false
+				changed, err = o.applyRemoteConfig(rc)
```

Comment:
looks like alot of the ApplyRemote tests are repeated. i feel like the test should look something like this (didn't include all the different config yaml files and a little more adjustments) 

```
func TestApplyRemoteConfig(t *testing.T) {
    tests := []struct {
        name           string
        file           string
        acceptsRemote  bool
        expectChanged  bool
        expectError    bool
        errorMessage   string
    }{
        {"Valid Config", "testdata/opamp.d/opamp-remote-config.yaml", true, true, false, ""},
        {"Does Not Accept Remote", "testdata/opamp.d/opamp-apache-config.yaml", false, false, true, "OpAMP agent does not accept remote configuration"},
        {"Invalid Config", "testdata/opamp.d/opamp-invalid-remote-config.yaml", true, false, true, "'max_elapsed_time' must be non-negative"},
    }

    for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            d, err := os.MkdirTemp("", "opamp.d")
            assert.NoError(t, err)
            defer os.RemoveAll(d)

            cfg := createDefaultConfig().(*Config)
            cfg.RemoteConfigurationDirectory = d
            set := extensiontest.NewNopSettings()
            o, err := newOpampAgent(cfg, set.Logger, set.BuildInfo, set.Resource)
            assert.NoError(t, err)

            cfg.AcceptsRemoteConfiguration = tt.acceptsRemote

            path := filepath.Join(tt.file)
            rb, err := os.ReadFile(path)
            assert.NoError(t, err)

            rc := &protobufs.AgentRemoteConfig{
                Config: &protobufs.AgentConfigMap{
                    ConfigMap: map[string]*protobufs.AgentConfigFile{
                        "default": {
                            Body: rb,
                        },
                    },
                },
                ConfigHash: []byte("b2b1e3e7f45d564db1c0b621bbf67008"),
            }

            changed, err := o.applyRemoteConfig(rc)
            if tt.expectError {
                assert.Error(t, err)
                assert.ErrorContains(t, err, tt.errorMessage)
            } else {
                assert.NoError(t, err)
            }
            assert.Equal(t, tt.expectChanged, changed)
        })
    }
}
```
so now the code isn't as repeated :D